CppTicTacToe/tictactoe.cpp

```diff
@@ -13,16 +13,18 @@ using namespace std;
 
 // main global variables and initialization follow here
 
+// Tony Comment
 int round_counter = 1;
 char mark = 'X';
 bool game_over = false;
 int play_again_input = 0;
 int spot_to_fill = 0; // for empty spots, until filled by player and used by player_move() and attempt_to_fill_spot()
+// Tony Comment
 char board_spots[9] = {'1', '2', '3', '4', '5', '6', '7', '8', '9'};
 
 
 // function prototypes follow here
-
+// Tony Comment
 void draw_board();
```

Comment:
Declaring all your function declarations here shows a lot about your code. Every function accepts no parameters and returns nothing. This makes things very hard to follow and reason about, even for you, the author.

Lets talk about the function declarations themselves. You dont actually need these. A function declaration is used as a place holder in c++. It basically says "im going to make this function later, i promise, just let me use it for now."

The compiler in C++ scans our code linearly. If it finds you try to use a function before its been declared, it will error. If you make a function declaration, and then make the function after you use it, that is allowed. It is, generally, bad practice to do so (unless you have a good reason).